torch/_dynamo/trace_rules.py

```diff
@@ -35,40 +40,2732 @@
 and trace_rules.lookup consolidation is done. Then the explicit listing of skip/inline objects have
 a higher priority, which can be used to override the skipfiles.check rules in some cases.
 """
-torch_name_rule_map = {
+manual_torch_name_rule_map = {
+    "torch.default_generator#get_state": TorchInGraphFunctionVariable,
+    "torch._C.Generator#get_state": TorchInGraphFunctionVariable,
+    "torch.default_generator#set_state": TorchInGraphFunctionVariable,
+    "torch._C.Generator#set_state": TorchInGraphFunctionVariable,
+    "torch.onnx.is_in_onnx_export": TorchInGraphFunctionVariable,
+    "torch.onnx.operators.shape_as_tensor": TorchInGraphFunctionVariable,
+    "torch.overrides.is_tensor_like": TorchInGraphFunctionVariable,
+    "torch.jit.is_scripting": TorchInGraphFunctionVariable,
+    "torch.jit.is_tracing": TorchInGraphFunctionVariable,
+    "torch.jit.annotate": TorchInGraphFunctionVariable,
+    "torch.distributed.is_available": TorchInGraphFunctionVariable,
+    "torch.distributed.is_initialized": TorchInGraphFunctionVariable,
+    "torch.distributed.get_rank": TorchInGraphFunctionVariable,
+    "torch.distributed.get_world_size": TorchInGraphFunctionVariable,
+    "torch.distributed._tensor.DTensor#from_local": TorchInGraphFunctionVariable,
+    "torch._utils.is_compiling": TorchInGraphFunctionVariable,
+    "torch.overrides.get_default_nowrap_functions": TorchInGraphFunctionVariable,
+    "torch.fx._symbolic_trace.is_fx_tracing": TorchInGraphFunctionVariable,
+    "torch._dynamo.external_utils.is_compiling": TorchInGraphFunctionVariable,
+}
+
+
+auto_torch_name_rule_map = {
```

Comment:
Is this auto-generated?  If so include a docstring with the needed commands to regenerate it.

Reply:
I think @yanboliang's intention was that we would initially generate this, but then as the library evolves, we would manually update the list (so this would be the source of truth).

Seeing this here, there are a few questions that pop immediately to mind. The biggest one is how we are going to keep this list up-to-date with `native_functions.yaml` changes. A very common thing is that someone adds a new operator to `native_functions.yaml`, and then, under the changes here, this map will have to be updated. This seems suboptimal. In particular, for operators that are C bindings into the dispatcher, I think there's a case to be made that we should *not* have to manually update them.

One non-problem is what to do about Python function implementations in PyTorch core. I think the intention here is that, when someone adds a new Python function, our default assumption should be that we will Dynamo trace into it, not allow it in the graph. I think it'd be OK to argue about this though!

Reply:
The ```auto_torch_name_rule_map``` is generated by the heuristic defined in ```test_trace_rules.py```, we will iterate from here, e.g, we have to remove the functions which should not be here, and move them to ```test_trace_rules.py:ignored_torch_name_rule_set``` to make the whole set on par with the set generated by the heuristic.
I think the ```test_trace_rules.py:test_torch_name_rule_map``` helped us to monitor changes happened inside of torch though it's not 100% guaranteed, but it's the best effort we can have.
Regarding to changes in ```native_functions.yaml```, can I say all functions in ```native_functions.yaml``` are in_graph_functions? If yes, probably I can add another rules to automatically update this map according the changes in ```native_functions.yaml```.

Reply:
Yes, everything `native_functions.yaml` is fine.

This is a little more complicated, though, because there is sometimes a nontrivial mapping from what is in native_functions.yaml and what actually pops up in the Python binding. This could easily end up being an entire PR's worth of work on its own.